frontend/src/main/scala/bloop/engine/BuildLoader.scala

```diff
@@ -80,16 +165,60 @@ object BuildLoader {
   def loadSynchronously(
       configDir: AbsolutePath,
       logger: Logger
-  ): List[Project] = {
+  ): List[LoadedProject] = {
+    val settings = WorkspaceSettings.readFromFile(configDir, logger)
     val configFiles = readConfigurationFilesInBase(configDir, logger).map { ap =>
       val bytes = ap.path.readAllBytes
       val hash = ByteHasher.hashBytes(bytes)
       Build.ReadConfiguration(Origin(ap, hash), bytes)
     }
 
     logger.debug(s"Loading ${configFiles.length} projects from '${configDir.syntax}'...")(
-      DebugFilter.Compilation
+      DebugFilter.All
     )
-    configFiles.map(f => Project.fromBytesAndOrigin(f.bytes, f.origin, logger))
+
+    configFiles.map { f =>
+      val project = loadProject(f.bytes, f.origin, logger)
+      settings match {
+        case None => LoadedProject.RawProject(project)
+        case Some(settings) =>
+          project.scalaInstance match {
+            case None => LoadedProject.RawProject(project)
+            case Some(instance) =>
+              def enableMetals(plugin: Option[AbsolutePath]) = {
+                LoadedProject.ConfiguredProject(
+                  Project.enableMetalsSettings(project, configDir, plugin, logger),
+                  project,
+                  settings
+                )
+              }
+
+              val scalaVersion = instance.version
```

Comment:
Since all paths end with metalsEnable, can't we put this into a function at least:
```
def enablePlugin[T](settingsEnable: Option[Path] => T): T 
```
a bit less of a duplciated code.